### PR TITLE
adding-wrap-doc-method-15

### PR DESCRIPTION
**Pull request by rahul799**
# Description

Please include a summary of the change and which issue is fixed.

# Checklist:
- [ ] Make sure that you are listed in the AUTHORS file
- [ ] Add relevant changes and new features to the CHANGELOG file
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] New and existing unit tests pass locally with my changes
- [ ] Updated or added python bindings for changed or new classes. (Tick if no updates were necessary.)

# How can I get additional information on failed tests during CI:
If your PR is failing you can check out 
- http://cdash.openms.de/index.php?project=OpenMS and look for your PR. If you click in the column that lists the failed tests you will get detailed error messages.

# Note:
- Once you opened a PR try to minimize the number of *pushes* to it as every push will trigger CI (automated builds and test) and is rather heavy on our infrastructure (e.g., if several pushes per day are performed).

### CODE REVIEW

**Comment by rahul799** [comment]
rebuild jenkins

File: src/pyOpenMS/pxds/Sample.pxd [hunk 1]
```diff
@@ -23,49 +23,49 @@ cdef extern from "<OpenMS/METADATA/Sample.h>" namespace "OpenMS":
...
-        double getConcentration() nogil except +
-        # sets the concentration (in g/l)
-        void setConcentration(double concentration) nogil except +
-
-        # returns a reference to the vector of subsamples that were combined to create this sample
-        libcpp_vector[Sample] getSubsamples() nogil except +
-        # sets the vector of subsamples that were combined to create this sample
-        void setSubsamples(libcpp_vector[Sample] subsamples) nogil except +
+        
+        String getNumber() nogil except + # wrap-doc:Returns the sample number (default: "")
```
[the remark this answers is not in the document]
Comment (oliveralka) [reply src/pyOpenMS/pxds/Sample.pxd]: @rahul799 This is a problem for the whole file/files - could you please take another look. It seems there are still occurences in this PR.

File: src/pyOpenMS/pxds/Sample.pxd [hunk 2]
```diff
@@ -23,49 +23,49 @@ cdef extern from "<OpenMS/METADATA/Sample.h>" namespace "OpenMS":
...
-        libcpp_vector[Sample] getSubsamples() nogil except +
-        # sets the vector of subsamples that were combined to create this sample
-        void setSubsamples(libcpp_vector[Sample] subsamples) nogil except +
+        
+        String getNumber() nogil except + # wrap-doc:Returns the sample number
+        
+        void setNumber(String number) nogil except + # wrap-doc:Sets the sample number (e.g. sample ID)
+
+        
+        String getComment() nogil except + # wrap-doc:Returns the comment (default: "")
```
Comment (oliveralka) [inline src/pyOpenMS/pxds/Sample.pxd:32]:
```suggestion
        String getComment() nogil except + # wrap-doc:Returns the comment (default "")
```

File: src/pyOpenMS/pxds/Sample.pxd [hunk 3]
```diff
@@ -23,49 +23,49 @@ cdef extern from "<OpenMS/METADATA/Sample.h>" namespace "OpenMS":
...
+        
+        void setComment(String comment) nogil except + # wrap-doc:Sets the comment (may contain newline characters)
+
+        
+        SampleState getState() nogil except + # wrap-doc:Returns the state of aggregation (default: SAMPLENULL)
```
Comment (oliveralka) [inline src/pyOpenMS/pxds/Sample.pxd:37]:
```suggestion
        SampleState getState() nogil except + # wrap-doc:Returns the state of aggregation (default SAMPLENULL)
```

File: src/pyOpenMS/pxds/Sample.pxd [hunk 4]
```diff
@@ -23,49 +23,49 @@ cdef extern from "<OpenMS/METADATA/Sample.h>" namespace "OpenMS":
...
+        
+        void setState(SampleState state) nogil except + # wrap-doc:Sets the state of aggregation
+
+        
+        double getMass() nogil except + # wrap-doc:Returns the mass (in gram) (default: 0.0)
```
Comment (oliveralka) [inline src/pyOpenMS/pxds/Sample.pxd:42]:
```suggestion
        double getMass() nogil except + # wrap-doc:Returns the mass (in gram) (default 0.0)
```

File: src/pyOpenMS/pxds/Sample.pxd [hunk 5]
```diff
@@ -23,49 +23,49 @@ cdef extern from "<OpenMS/METADATA/Sample.h>" namespace "OpenMS":
...
+        
+        void setMass(double mass) nogil except + # wrap-doc:Sets the mass (in gram)
+
+        
+        double getVolume() nogil except + # wrap-doc:Returns the volume (in ml) (default: 0.0)
```
Comment (oliveralka) [inline src/pyOpenMS/pxds/Sample.pxd:47]:
```suggestion
        double getVolume() nogil except + # wrap-doc:Returns the volume (in ml) (default 0.0)
```

File: src/pyOpenMS/pxds/Sample.pxd [hunk 6]
```diff
@@ -23,49 +23,49 @@ cdef extern from "<OpenMS/METADATA/Sample.h>" namespace "OpenMS":
...
+        
+        void setVolume(double volume) nogil except + # wrap-doc:Sets the volume (in ml)
+
+        
+        double getConcentration() nogil except + # wrap-doc:Returns the concentration (in g/l) (default: 0.0)
```
Comment (oliveralka) [inline src/pyOpenMS/pxds/Sample.pxd:52]:
```suggestion
        double getConcentration() nogil except + # wrap-doc:Returns the concentration (in g/l) (default 0.0)
```

File: src/pyOpenMS/pxds/SemanticValidator.pxd [hunk 7]
```diff
@@ -18,32 +18,23 @@ cdef extern from "<OpenMS/FORMAT/VALIDATORS/SemanticValidator.h>" namespace "Ope
 
         bool validate(String filename, StringList errors, StringList warnings) nogil except +
 
-        # Checks if a CVTerm is allowed in a given path
-        bool locateTerm(String path, SemanticValidator_CVTerm & parsed_term) nogil except +
+        bool locateTerm(String path, SemanticValidator_CVTerm & parsed_term) nogil except + # wrap-doc:Checks if a CVTerm is allowed in a given path
 
-        # Sets the CV parameter tag name (default: 'cvParam')
-        void setTag(String tag) nogil except +
+        void setTag(String tag) nogil except + # wrap-doc:Sets the CV parameter tag name (default: 'cvParam')
```
Comment (oliveralka) [inline src/pyOpenMS/pxds/SemanticValidator.pxd:23]:
```suggestion
        void setTag(String tag) nogil except + # wrap-doc:Sets the CV parameter tag name (default 'cvParam')
```

File: src/pyOpenMS/pxds/SemanticValidator.pxd [hunk 8]
```diff
@@ -18,32 +18,23 @@ cdef extern from "<OpenMS/FORMAT/VALIDATORS/SemanticValidator.h>" namespace "Ope
...
 
-        # Sets the name of the attribute for accessions in the CV parameter tag name (default: 'accession')
-        void setAccessionAttribute(String accession) nogil except +
+        void setAccessionAttribute(String accession) nogil except + # wrap-doc:Sets the name of the attribute for accessions in the CV parameter tag name (default: 'accession')
```
Comment (oliveralka) [inline src/pyOpenMS/pxds/SemanticValidator.pxd:25]:
```suggestion
        void setAccessionAttribute(String accession) nogil except + # wrap-doc:Sets the name of the attribute for accessions in the CV parameter tag name (default 'accession')
```

File: src/pyOpenMS/pxds/SemanticValidator.pxd [hunk 9]
```diff
@@ -18,32 +18,23 @@ cdef extern from "<OpenMS/FORMAT/VALIDATORS/SemanticValidator.h>" namespace "Ope
...
 
-        # Sets the name of the attribute for accessions in the CV parameter tag name (default: 'name')
-        void setNameAttribute(String name) nogil except +
+        void setNameAttribute(String name) nogil except + # wrap-doc:Sets the name of the attribute for accessions in the CV parameter tag name (default: 'name')
```
Comment (oliveralka) [inline src/pyOpenMS/pxds/SemanticValidator.pxd:27]:
```suggestion
        void setNameAttribute(String name) nogil except + # wrap-doc:Sets the name of the attribute for accessions in the CV parameter tag name (default 'name')
```

File: src/pyOpenMS/pxds/SemanticValidator.pxd [hunk 10]
```diff
@@ -18,32 +18,23 @@ cdef extern from "<OpenMS/FORMAT/VALIDATORS/SemanticValidator.h>" namespace "Ope
...
 
-        # Sets the name of the attribute for accessions in the CV parameter tag name (default: 'value')
-        void setValueAttribute(String value) nogil except +
+        void setValueAttribute(String value) nogil except + # wrap-doc:Sets the name of the attribute for accessions in the CV parameter tag name (default: 'value')
```
Comment (oliveralka) [inline src/pyOpenMS/pxds/SemanticValidator.pxd:29]:
```suggestion
        void setValueAttribute(String value) nogil except + # wrap-doc:Sets the name of the attribute for accessions in the CV parameter tag name (default 'value')
```

File: src/pyOpenMS/pxds/SemanticValidator.pxd [hunk 11]
```diff
@@ -18,32 +18,23 @@ cdef extern from "<OpenMS/FORMAT/VALIDATORS/SemanticValidator.h>" namespace "Ope
...
-        void setCheckTermValueTypes(bool check) nogil except +
+        void setCheckTermValueTypes(bool check) nogil except + # wrap-doc:Set if CV term value types should be check (enabled by default)
 
-        # Set if CV term units should be check (disabled by default)
-        void setCheckUnits(bool check) nogil except +
+        void setCheckUnits(bool check) nogil except + # wrap-doc:Set if CV term units should be check (disabled by default)
 
-        # Sets the name of the unit accession attribute (default: 'unitAccession')
-        void setUnitAccessionAttribute(String accession) nogil except +
+        void setUnitAccessionAttribute(String accession) nogil except + # wrap-doc:Sets the name of the unit accession attribute (default: 'unitAccession')
```
Comment (oliveralka) [inline src/pyOpenMS/pxds/SemanticValidator.pxd:35]:
```suggestion
        void setUnitAccessionAttribute(String accession) nogil except + # wrap-doc:Sets the name of the unit accession attribute (default 'unitAccession')
```

File: src/pyOpenMS/pxds/SemanticValidator.pxd [hunk 12]
```diff
@@ -18,32 +18,23 @@ cdef extern from "<OpenMS/FORMAT/VALIDATORS/SemanticValidator.h>" namespace "Ope
...
 
-        # Sets the name of the unit name attribute (default: 'unitName')
-        void setUnitNameAttribute(String name) nogil except +
+        void setUnitNameAttribute(String name) nogil except + # wrap-doc:Sets the name of the unit name attribute (default: 'unitName')
```
Comment (oliveralka) [inline src/pyOpenMS/pxds/SemanticValidator.pxd:37]:
```suggestion
        void setUnitNameAttribute(String name) nogil except + # wrap-doc:Sets the name of the unit name attribute (default 'unitName')
```

File: src/pyOpenMS/pxds/SpectrumLookup.pxd [hunk 13]
```diff
@@ -12,26 +12,61 @@ cdef extern from "<OpenMS/METADATA/SpectrumLookup.h>" namespace "OpenMS":
...
         # allow inheritance of overloaded methods.
 
         # Size findByIndex(Size index) nogil except +
 
         Size findByIndex(Size index, bool count_from_one) nogil except +
+         # wrap-doc:
+                #   Look up spectrum by index (position in the vector of spectra)
+                #   -----
+                #   :param index: Index to look up
+                #   :param count_from_one: Do indexes start counting at one (default: zero)?
```
Comment (oliveralka) [inline src/pyOpenMS/pxds/SpectrumLookup.pxd:48]:
```suggestion
                #   :param count_from_one: Do indexes start counting at one (default zero)?
```

File: src/pyOpenMS/pxds/SequestOutfile.pxd [hunk 14]
```diff
@@ -22,13 +22,24 @@ cdef extern from "<OpenMS/FORMAT/SequestOutfile.h>" namespace "OpenMS":
...
+                #   -----
+                #   :param result_filename: The file to be loaded
+                #   :param peptide_identifications: The identifications
+                #   :param protein_identification: The protein identifications
+                #   :param p_value_threshold: The significance level (for the peptide hit scores)
+                #   :param pvalues: A list with the pvalues of the peptides (pvalues computed with peptide prophet)
+                #   :param database: The database used for the search
+                #   :param ignore_proteins_per_peptide: This is a hack to deal with files that use a suffix like "+1" in column "Reference", but do not actually list extra protein references in subsequent lines
+
+        bool getColumns(const String &line, libcpp_vector[ String ] &substrings, Size number_of_columns, Size reference_column) nogil except + retrieve columns from a Sequest outfile line
```
Comment (oliveralka) [inline src/pyOpenMS/pxds/SequestOutfile.pxd:36]:
```suggestion
        bool getColumns(const String &line, libcpp_vector[ String ] &substrings, Size number_of_columns, Size reference_column) nogil except + # wrap-doc:Retrieve columns from a Sequest outfile line
```

**Review by oliveralka — APPROVED** [review]
Looks good - thanks!

**Pull request merged**